client/app/bundles/course/lesson_plan/actions.jsx

```diff
@@ -7,3 +7,101 @@ export function toggleItemTypeVisibility(itemType) {
     itemType,
   };
 }
+
+function combineDateTime(dateSource, timeSource) {
+  const combinedDateTime = dateSource || new Date();
+  if (timeSource) {
+    combinedDateTime.setHours(timeSource.getHours());
+    combinedDateTime.setMinutes(timeSource.getMinutes());
+  }
+  return combinedDateTime;
+}
+
+function patchRequest(endpoint, payload, successHandler, failureHandler) {
+  const csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+  const headers = { Accept: 'application/json' };
+  axios.patch(endpoint, { ...payload, authenticity_token: csrfToken }, { headers })
+    .then(successHandler)
+    .catch(failureHandler);
+}
+
+export function updateItemField(id, field, value, oldValue) {
+  const payload = {
+    item: {
+      [field]: value,
+    },
+  };
+
+  return (dispatch) => {
+    dispatch({
+      type: actionTypes.SET_ITEM_FIELD,
```

Comment:
I feel the action name here a bit confusing. How about we scandalise the server related action names to:
```javascript
ACTION_PENDING // or ACTION_START 
ACTION_FULFILLED // or ACTION_FINISHED
ACTION_REJECTED // or ACTION_FAILED
```

Just like what `redux-promise-middleware` did: https://github.com/pburtchaell/redux-promise-middleware/blob/master/docs/guides/optimistic-updates.md

So in `PENDING` action we can update the UI optimistically already